### PR TITLE
Pagination Refactor

### PR DESCRIPTION
**Pull request by thostetler**
Changes pagination component and hook
Fixes several issues
De-couples hook from the Pagination component

image
image
image

### CODE REVIEW

**Comment by thostetler** [comment]
@shinyichen hold off just a little bit on doing a full review, small thing for me to add.

**Comment by thostetler** [comment]
@shinyichen this is ready to go (I hope 🤞)
let me know how it goes
thanks!

File: src/pages/search/index.tsx [hunk 1]
```diff
@@ -121,20 +122,44 @@ const SearchPage: NextPage<ISearchPageProps> = ({ searchParams }) => {
...
+          {isLoading && <LoadingList />}
+          {isSuccess && !isLoading && <SimpleResultList docs={data.docs} indexStart={pagination.startIndex} />}
+          {isSuccess && !isLoading && <Pagination totalResults={data.numFound} {...pagination} />}
         </Box>
       </form>
     </Box>
   );
 };
 
+const LoadingList = () => {
```
Comment (shinyichen) [inline src/pages/search/index.tsx:134]: We do have a `ItemsSkeleton` component

**Review by shinyichen — APPROVED** [review]
Just the couple of comments.

**Comment by thostetler** [comment]
@shinyichen good call, made both suggested changes!

**Pull request merged**